### PR TITLE
Added support for the Arduino Nano ESP32

### PR DESCRIPTION
**Pull request by spierepf**
I noticed that there was no config for the new Nano ESP32, so I added a stanza to `platformio.ini`. Let me know what you think.

### CODE REVIEW

**Comment by blazoncek** [comment]
Thank you.
What about other build flags? Don't need any?

**Comment by spierepf** [comment]
No other build flags were required for me to build the code and upload it to my Arduino ESP32 Nano.

**Comment by softhack007** [comment]
Actually it might compile, but it's necessary to also include all other build flags for -S3. Additionally directly using a git reference for the platform is not a good idea.

You might want to look at [env:esp32s3dev_8MB] for an example of a complete buildenv. Please add at least `build_flags` and `lib_deps`.

File: platformio.ini [hunk 1]
```diff
@@ -805,3 +806,13 @@ lib_deps =
   ${esp32.lib_deps}
   TFT_eSPI @ ^2.3.70
 board_build.partitions = ${esp32.default_partitions}
+
+# ------------------------------------------------------------------------------
+# Arduino® Nano ESP32
+# ------------------------------------------------------------------------------
+[env:arduino_nano_esp32]
+platform = https://github.com/platformio/platform-espressif32.git
```
Comment (softhack007) [inline platformio.ini:814]: should be `platform = espressif32@6.4.0`

File: platformio.ini [hunk 2]
```diff
@@ -805,3 +806,13 @@ lib_deps =
...
+board = arduino_nano_esp32
+upload_protocol = esptool
+board_build.partitions = tools/WLED_ESP32_16MB.csv
+board_build.flash_mode = dio
```
Comment (softhack007) [inline platformio.ini:818]: For proper builds, please add at least the following parts from [[env:esp32s3dev_8MB_PSRAM_opi]](https://github.com/Aircoookie/WLED/blob/4245767357dc609fa1f567b22a0817ddc3e3971e/platformio.ini#L507)
* `extends`
* `build_flags`
* `lib_deps`

Maybe you also need `build_unflags`, if some standard  USB flags are in conflict to defaults for your board.

File: platformio.ini [hunk 3]
```diff
@@ -805,3 +806,13 @@ lib_deps =
   ${esp32.lib_deps}
   TFT_eSPI @ ^2.3.70
 board_build.partitions = ${esp32.default_partitions}
+
+# ------------------------------------------------------------------------------
```
Comment (softhack007) [inline platformio.ini:810]: Actually one or two headlines should be enough ;-)

Comment (softhack007) [reply platformio.ini:810]: Just saw that the previous entry also uses a FAT headline. So OK for me if you do the same.

File: platformio.ini (hunk 1 above)
Comment (spierepf) [reply platformio.ini:814]: I tried that, but it causes the build to fail, looking for something called `FrameworkArduinoVariant`.

**Comment by blazoncek** [comment]
> No other build flags were required for me to build the code and upload it to my Arduino ESP32 Nano.

It is not about *required* flags but about *optimal* flags.

File: platformio.ini [hunk 4]
```diff
@@ -805,3 +806,21 @@ lib_deps =
...
+[env:arduino_nano_esp32]
+platform = https://github.com/platformio/platform-espressif32.git
+board = arduino_nano_esp32
+upload_protocol = esptool
+board_build.partitions = tools/WLED_ESP32_16MB.csv
+board_build.flash_mode = qio
+build_unflags = ${common.build_unflags}
+build_flags = ${common.build_flags} ${esp32s3.build_flags}
+  -D CONFIG_LITTLEFS_FOR_IDF_3_2 -D WLED_WATCHDOG_TIMEOUT=0
```
Comment (blazoncek) [inline platformio.ini:821]: Are you sure this one is optimal for latest core? Arduino core in platform 4 and above have LittleFS included (IDF v4).

**Comment by softhack007** [comment]
@spierepf please check if this buildenv works for you. I could compile WLED with it - avoiding the .git platform reference.

```ini
# ------------------------------------------------------------------------------
# Arduino(R) Nano ESP32
# ------------------------------------------------------------------------------
[env:arduino_nano_esp32]
;;platform = https://github.com/platformio/platform-espressif32.git
platform = espressif32@ ~6.4.0
platform_packages =
board = arduino_nano_esp32
upload_protocol = esptool
board_build.partitions = tools/WLED_ESP32_16MB.csv
board_build.flash_mode = qio
build_unflags = ${common.build_unflags}
build_flags = ${common.build_flags} ${esp32s3.build_flags}
  ;-D ARDUINO_USB_MODE=0 ;; Normal mode (TinyUSB)
  -D ARDUINO_USB_CDC_ON_BOOT=1 -D ARDUINO_USB_MODE=1 ;; Debug mode (Hardware CDC)
  ;-DBOARD_USES_HW_GPIO_NUMBERS ;; for pin numbers By GPIO numbers
  ;-D WLED_RELEASE_NAME=ESP32-S3_PSRAM
  -D CONFIG_LITTLEFS_FOR_IDF_3_2 -D WLED_WATCHDOG_TIMEOUT=0 ;; legacy flags
  -D WLED_USE_PSRAM -DBOARD_HAS_PSRAM ; tells WLED that PSRAM shall be used
lib_deps = ${esp32s3.lib_deps}

```

File: platformio.ini (hunk 4 above)
Comment (softhack007) [reply platformio.ini:821]: not 100% sure its needed - build-in LittleFS is always used, this flag is also there in other -S3 envs - maybe could be omitted.

File: platformio.ini (hunk 1 above)
Comment (spierepf) [reply platformio.ini]: Apparently, requesting two builds back to back resolves the `FrameworkArduinoVariant` error. Odd.

File: platformio.ini (hunk 2 above)
Comment (spierepf) [reply platformio.ini]: Done

File: platformio.ini (hunk 4 above)
Comment (spierepf) [reply platformio.ini]: I've removed `-D CONFIG_LITTLEFS_FOR_IDF_3_2`.

**Comment by spierepf** [comment]
I have made changes to `platform` and removed `-D CONFIG_LITTLEFS_FOR_IDF_3_2` as requested. @softhack007 suggested buildenv built successfully, but would not upload to my device.

**Comment by softhack007** [comment]
Did a bit of research, and it seems that platformio support for this board is still "bleeding edge". It would be more stable with arduino-esp32 v2.0.14, however platformio registry on goes up to v2.0.11. 
So if the board basically work with platform 6.4.0, lets go with that one.

<br/>

This is the latest tools & packages coming with platform `dev` version
```log
Processing arduino_nano_esp32 (platform: https://github.com/platformio/platform-espressif32.git; board: arduino_nano_esp32; framework: arduino)
PLATFORM: Espressif 32 (6.4.0+sha.f6ec392) > Arduino Nano ESP32
   HARDWARE: ESP32S3 240MHz, 320KB RAM, 16MB Flash
PACKAGES:
 - framework-arduinoespressif32 @ 3.20011.230801 (2.0.11)
 - tool-dfuutil-arduino @ 1.11.0
 - tool-esptoolpy @ 1.40501.0 (4.5.1)
 - tool-openocd-esp32 @ 2.1100.20220706 (11.0)
 - toolchain-riscv32-esp @ 8.4.0+2021r2-patch5
 - toolchain-xtensa-esp32s3 @ 8.4.0+2021r2-patch5
 ```

This is the tools when using latest `release` platform 6.4.0
```log
Processing arduino_nano_esp32 (platform: espressif32@ ~6.4.0; board: arduino_nano_esp32; framework: arduino)
PLATFORM: Espressif 32 (6.4.0) > Arduino Nano ESP32
   HARDWARE: ESP32S3 240MHz, 320KB RAM, 16MB Flash
PACKAGES:
 - framework-arduinoespressif32 @ 3.20011.230801 (2.0.11)
 - tool-dfuutil-arduino @ 1.11.0
 - tool-esptoolpy @ 1.40501.0 (4.5.1)
 - tool-openocd-esp32 @ 2.1100.20220706 (11.0)
 - toolchain-riscv32-esp @ 8.4.0+2021r2-patch5
 - toolchain-xtensa-esp32s3 @ 8.4.0+2021r2-patch5
```

File: platformio.ini [hunk 5]
```diff
@@ -805,3 +806,21 @@ lib_deps =
   ${esp32.lib_deps}
   TFT_eSPI @ ^2.3.70
 board_build.partitions = ${esp32.default_partitions}
+
+# ------------------------------------------------------------------------------
+# Arduino® Nano ESP32
```
Comment (softhack007) [inline platformio.ini:811]: just one last (tiny) request, as I would like to keep the file "pure ASCII".
Please change "Arduino®" to "Arduino(R)"

**Review by softhack007 — CHANGES_REQUESTED** [review]

File: platformio.ini (hunk 5 above)
Comment (spierepf) [reply platformio.ini]: Done.

**Review by softhack007 — APPROVED** [review]
ok from my side, thanks 👍 
cound be merged directly after 0.14.0 was released.

File: platformio.ini [hunk 6]
```diff
@@ -805,3 +806,21 @@ lib_deps =
   ${esp32.lib_deps}
   TFT_eSPI @ ^2.3.70
 board_build.partitions = ${esp32.default_partitions}
+
+# ------------------------------------------------------------------------------
+# Arduino(R) Nano ESP32
+# ------------------------------------------------------------------------------
+[env:arduino_nano_esp32]
+platform = espressif32@ ~6.4.0
```
Comment (softhack007) [inline platformio.ini:814]: Sorry, overlooked one.
Please add this line, directly after the platform= line:

```ini
platform_packages =
```

Reason is that our [common] section defines platform packages for 8266. So as best practice, this variable should be cleared on esp32.

Comment (spierepf) [reply platformio.ini:814]: Done.

**Comment by Aircoookie** [comment]
Thank you for the contribution!
I can't properly rebase this, so it might be easiest if you make a new PR to the `0_15` branch.

I also discussed with @blazoncek leaving only a bare-minimum environment per supported platform (basically just the release environments for 8266, 32, S2, S3, C3) and moving all other board-specific environments into a separate .ini file, but that will be done in a separate PR, so adding this change to `platformio.ini` is fine for the time being.

**Comment by blazoncek** [comment]
We will not be adding any more custom or specialized environments to `platformio.ini`.
You are encouraged to make a new PR with changes (added environment) in `platformio_override.sample.ini` (based on 0_15 branch ATM).

**Pull request closed without merging**